src/lib/import_export/binary/binary_parser.cpp

```diff
@@ -202,6 +204,27 @@ std::shared_ptr<DictionarySegment<T>> BinaryParser::_import_dictionary_segment(s
   return std::make_shared<DictionarySegment<T>>(dictionary, attribute_vector);
 }
 
+template <typename T>
+std::shared_ptr<VariableStringDictionarySegment<T>> BinaryParser::_import_variable_string_length_segment(
+    std::ifstream& file, ChunkOffset row_count) {
+
+  // Read attribute vector compression type and use it to decompress.
+  const auto compressed_vector_type_id = _read_value<CompressedVectorTypeID>(file);
+  auto attribute_vector = _import_attribute_vector(file, row_count, compressed_vector_type_id);
+
+  // Read offset vector.
+  const auto offset_vector_size = _read_value<uint32_t>(file);
+  auto offset_vector = _read_values<uint32_t>(file, offset_vector_size);
+
+  // Read dictionary.
+  const auto dictionary_size = _read_value<uint32_t>(file);
+  auto dictionary = _read_values<char>(file, dictionary_size);
```

Comment:
```suggestion
  const auto dictionary = _read_values<char>(file, dictionary_size);
```

Reply:
And above, I guess. Does not really add anything, but oh well, contributing.md kind of demands it.